lib/plugins/aws/lib/naming.js

```diff
@@ -448,7 +448,11 @@ module.exports = {
       multiValueHeaders ? 'multi-value-' : ''
     }${this.provider.getStage()}`;
   },
-  getAlbTargetGroupName(functionName, albId, multiValueHeaders) {
+  getAlbTargetGroupName(targetGroupName, functionName, albId, multiValueHeaders) {
+    if (targetGroupName) {
+      return this.provider.getAlbTargetGroupPrefix() + targetGroupName;
```

Comment:
I believe that in that case we should not rely on `provider.alb.targetGroupPrefix` at all.

This property was introduced to have some constant prefix in names which represent md5 hashes, as with md5 hashes, even small change in string we hash, results in completely different hash (so e.g. `foo-dev` and `foo-prod` will result with completely different names, so even though service name didn't change, we receive no common prefix for both names).

Now, when we explicitly set the name, there's no longer this problem, and I believe there's no use case here to rely on `provider.alb.targetGroupPrefix`  (we just probably should add in documentation that this setting is exclusive to `functions[].events[].alb.targetGroupName`)

Ideally we should also throw if both are used.

Reply:
Indeed I wasn't sure which was the best behavior, I will change as you suggest !